### PR TITLE
fix bug with setting android id

### PR DESCRIPTION
**Pull request by chadkouse**
I found that in the `_oauth` function it is sending a blank `androidId`.. See https://github.com/jamon/playmusic/blob/master/play.js#L151

the variable `_androidId` is never set

Yesterday google started rejecting my requests..  Even changing my password, etc didn't help.  Once I added this line it started working 😩

### CODE REVIEW

**Comment by chadkouse** [comment]
@jamon This may be a red herring -- now I'm getting sporadic failures...

File: play.js [hunk 1]
```diff
@@ -181,6 +181,7 @@ PlayMusic.prototype._oauth =  function (callback) {
 PlayMusic.prototype.login =  function (opt, callback) {
     var that = this;
     opt.androidId = opt.androidId || crypto.pseudoRandomBytes(8).toString("hex");
+    that._androidId = opt.androidId;
```
Comment (kweiberth) [inline play.js]: @jamon @chadkouse I'm seeing this exact issue at the line you linked to: https://github.com/jamon/playmusic/blob/master/play.js#L151. 

However, I don't think this solution will fix the problem (and this may be why you're seeing failures again?). You're setting `_androidId` onto `that`, but `that` is a `var` that is scoped to this method (`PlayMusic.prototype.login`) and won't be picked up by the `_oauth` method. A better solution might be to add `crypto.pseudoRandomBytes(8).toString("hex")` right in `_oauth` on https://github.com/jamon/playmusic/blob/master/play.js#L151. I think this is what we want since `_oauth` is called only from `init`, which doesn't accept an `androidId`, so we have to supply one.

**Comment by kweiberth** [comment]
I also have been encountering sporadic failures as well.

File: play.js (hunk 1 above)
Comment (chadkouse) [inline play.js]: my fix did actually pick up the androidId that was set in the _oauth method fwiw.  `this` is not scoped to the method but to the PlayMusic prototype class...

Comment (kweiberth) [inline play.js]: you might be right. To me, it looks like `_oauth` method needs `_androidId` on `that`, which comes right from `this`. I see `_oauth` being called only from `init` method, which does not set `_androidId` on anything. In your fix, I see you grabbing `androidId` from the arguments of `login` or creating it fresh, then setting it on `that`, which to me seems like a method-scoped variable and I don't know how `_oauth` will pick it up.

A bigger issue might be the intermittent failures. I'm definitely seeing lots of errors. I know this is an unofficial API... is there an "unofficial" contact from Google who can help us out?

**Comment by chadkouse** [comment]
@kweiberth I ran the oauth call directly (using postman) and got my auth token out of there.. then I pass that into the init method and set `this._token` to the token I got.

After that stuff started working reliably again.  I think google made a change to something and now they don't like me getting a new oauth token every request (which kinda makes sense) -- I think long term the token should be cached somehow (on disk?)

TL;DR I grabbed my actual androidId from my device and grabbed the oauth token I got from auth'ing against postman and put those into the init method of `play.js` and things started working.

**Comment by kweiberth** [comment]
Oh wow, thanks. That clarifies a lot for me. And I guess that makes sense. I think you're right: long term solution is to store the oauth token and reduce num of request.

**Comment by kweiberth** [comment]
@chadkouse how long have you been able to use the same auth token? Does it seem like they expire?

**Comment by FunkeDope** [comment]
I'm still not able to get this working and I'm not sure what I'm missing. I'm getting a 403 through Postman too.

I have 2fa on, however I've created a separate App Password which should allow this to generate a token. This is how the python library is working.

Any other ideas you guys can think of? This entire login situation needs a total rewrite.

**Comment by kweiberth** [comment]
I've been using the same auth token for a while now (since early June) and no issues. Seems like they don't expire? Not sure. Agree with you, the auth situation needs a rewrite. Honestly, it would be great if Google offered an official and supported API.
